src/overlays/actors/ovl_En_Heishi1/z_en_heishi1.c

```diff
@@ -21,37 +43,473 @@ const ActorInit En_Heishi1_InitVars = {
+                break;
+            case true:
+                if (this->headTimer == 0) {
+                    if (this->waitTimer == 0) {
+                        if ((this->type == 0) || (this->type == 1)) {
+                            this->waypoint++;
+                            if (this->waypoint >= 4) {
+                                this->waypoint = 0;
+                            }
+                        } else {
+                            // waypoints are defined with corners as 0-3 and middle points as 4-7
+                            // to choose the next waypoint they hardcoded the order "04152637" in an array
+                            for (i = 0; i < ARRAY_COUNT(sWaypoints); i++) {
+                                if (this->waypoint == sWaypoints[i]) {
+                                    i++;
+                                    if (i >= ARRAY_COUNT(sWaypoints)) {
+                                        i = 0;
+                                    }
+                                    this->waypoint = sWaypoints[i];
+                                    break;
+                                }
+                            }
+                            this->waypointTimer = 5;
+                        }
+                        this->actionFunc = EnHeishi1_SetupWalk;
+                    } else {
+                        this->headBehaviorDecided = false;
+                    }
+                }
+                break;
+        }
+        Math_SmoothScaleMaxF(&this->headAngle, this->headAngleTarget, this->headTurnSpeedScale,
+                             this->headTurnSpeedMax + this->headTurnSpeedMax);
+
+        if ((this->path == BREG(1)) && (BREG(0) != 0)) {
+            osSyncPrintf(VT_FGCOL(GREEN) " 種類  %d\n" VT_RST, this->path);
+            osSyncPrintf(VT_FGCOL(GREEN) " ぱす  %d\n" VT_RST, this->waypoint);
+            osSyncPrintf(VT_FGCOL(GREEN) " 反転  %d\n" VT_RST, this->bodyTurnSpeed);
+            osSyncPrintf(VT_FGCOL(GREEN) " 時間  %d\n" VT_RST, this->waypointTimer);
+            osSyncPrintf("\n\n");
+        }
+    }
+}
+
+void EnHeishi1_SetupTurnTowardLink(EnHeishi1* this, GlobalContext* globalCtx) {
+    s16 frameCount = (f32)SkelAnime_GetFrameCount(&D_06005C30.genericHeader);
+
+    SkelAnime_ChangeAnim(&this->skelAnime, &D_06005C30, 1.0f, 0.0f, frameCount, 0, -10.0f);
+    this->kickTimer = 30;
+    this->actionFunc = EnHeishi1_TurnTowardLink;
+}
+
+void EnHeishi1_TurnTowardLink(EnHeishi1* this, GlobalContext* globalCtx) {
+    SkelAnime_FrameUpdateMatrix(&this->skelAnime);
+
+    if (this->type != 5) {
+        Math_SmoothScaleMaxMinS(&this->actor.shape.rot.y, this->actor.yawTowardsLink, 3, this->bodyTurnSpeed, 0);
+        Math_SmoothScaleMaxF(&this->bodyTurnSpeed, 3000.0f, 1.0f, 300.0f);
+        Math_SmoothDownscaleMaxF(&this->headAngle, 0.5f, 2000.0f);
+    }
+
+    // wait 1.5 seconds after reaching link before allowing the text to be advanced
+    if (this->kickTimer == 0) {
```

Comment:
The comment about 1.5 seconds doesn't belong here imo, since it's not where the 30 frames are actually set.